test/controllers/register/SchemeAddressListControllerSpec.scala

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.register
+
+import connectors.FakeDataCacheConnector
+import controllers.ControllerSpecBase
+import controllers.actions._
+import forms.register.SchemeAddressListFormProvider
+import identifiers.register.SchemePostCodeLookupId
+import models.NormalMode
+import models.addresslookup.Address
+import models.register.SchemeDetails
+import models.register.SchemeType.SingleTrust
+import org.mockito.Matchers.any
+import org.mockito.Mockito.{times, verify}
+import org.mockito.{Matchers, Mockito}
+import play.api.data.Form
+import play.api.libs.json._
+import play.api.test.Helpers._
+import utils.FakeNavigator
+import views.html.register.schemeAddressList
+
+class SchemeAddressListControllerSpec extends ControllerSpecBase {
+
+  def onwardRoute = controllers.routes.IndexController.onPageLoad()
+
+  val formProvider = new SchemeAddressListFormProvider()
+  val schemeName = "ThisSchemeName"
+  val schemeDetails = Json.obj("schemeDetails" -> SchemeDetails(schemeName, SingleTrust))
+  val addresses = Seq(
+    address("test post code 1"),
+    address("test post code 2")
+  )
+  val addressObject = Json.obj(SchemePostCodeLookupId.toString -> addresses)
+
+  val dataCacheConnector = Mockito.spy(new FakeDataCacheConnector())
```

Comment:
please try not to use spy instead use mocking as its done in other pages AddressListControllerSpec